zopkio/adhoc_deployer.py

```diff
@@ -135,10 +135,33 @@ def install(self, unique_id, configs=None):
         if executable is None:
           logger.error("executable was not provided for unique_id: " + unique_id)
           raise DeploymentError("executable was not provided for unique_id: " + unique_id)
-        exec_name = os.path.basename(executable)
-        install_location = os.path.join(install_path, exec_name)
-        with get_sftp_client(hostname, username=runtime.get_username(), password=runtime.get_password()) as ftp:
-          ftp.put(executable, install_location)
+
+        #if the executable is in remote location copy to local machine
+        copy_from_remote_location = False;
```

Comment:
Ah.. I remember why I avoided this.. because after every download Zopkio will delete the file :) Should I change it to deleting if only Zopkio downloaded or drop the delete all together for the caching ??